sphereon-kmp-crypto-kms-azure/src/commonTest/kotlin/com/sphereon/crypto/kms/azure/AzureKeyVaultProviderTest.kt

```diff
@@ -0,0 +1,143 @@
+package com.sphereon.crypto.kms.azure
+
+import com.sphereon.crypto.generic.Curve
+import com.sphereon.crypto.generic.DigestAlg
+import com.sphereon.crypto.generic.KeyOperations
+import com.sphereon.crypto.generic.KeyType
+import com.sphereon.crypto.generic.SignatureAlgorithm
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.TestInstance
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class AzureKeyVaultProviderTest {
+    private lateinit var azureKeyVaultCryptoProvider: AzureKeyVaultCryptoProvider
+
+    @BeforeTest
+    fun setUp() {
+        val azureConfig = AzureKeyvaultClientConfig(
+            keyvaultUrl = BuildKonfig.AZURE_KEYVAULT_URL!!,
+            tenantId = BuildKonfig.AZURE_KEYVAULT_TENANT_ID!!,
+            credentialOpts = CredentialOpts(
+                credentialMode = CredentialMode.SERVICE_CLIENT_SECRET, // Use a client id and secret to authenticate as an app
+                secretCredentialOpts = SecretCredentialOpts(
+                    clientId = BuildKonfig.AZURE_KEYVAULT_CLIENT_ID!!,
+                    clientSecret = BuildKonfig.AZURE_KEYVAULT_CLIENT_SECRET!!
+                )
+            ),
+            hsmType = HSMType.KEYVAULT, // Either KEYVAULT as HSM (FIPS140 Level-2), or MANAGED_HSM
+            applicationId = "azure-keyvault-test", // This can be randomly choosen
+            exponentialBackoffRetryOpts = ExponentialBackoffRetryOpts(
+                maxRetries = 10, // let's try max 10 times
+                baseDelayInMS = 500, // Wait 0,5 seconds the first time
+                maxDelayInMS = 15000 // Wait for max 15 seconds eventually
+            )
+        )
+        azureKeyVaultCryptoProvider = AzureKeyVaultCryptoProvider(
+            id = "azure-keyvault-test",
+            config = azureConfig
+        )
+
+        azureKeyVaultCryptoProvider = AzureKeyVaultCryptoProvider(id = "test-azure-key-vault", config = azureConfig)
```

Comment:
Why are you overwriting the above property with one with a different name?

Reply:
Left over by mistake. Removed.